lib/apis/moray.js

```diff
@@ -524,7 +525,16 @@ Moray.prototype.listVms = function listVms(params, raw, cb) {
             if (object && object.value) {
                 vm  = (raw ? object.value
                            : common.translateVm(object.value, true));
-                vms.push(vm);
+                // The way the API works wrt markers is that the client
+                // requests the first page of results. Then to get the second
+                // page of results, the client adds a marker to the query
+                // string. This marker identifies the last entry of the first
+                // page. Thus, if a marker is used, do not include the VM that
+                // corresponds to that marker in the results of a given page,
+                // as the client already got it with the previous page.
+                if (!params.marker ||
+                    !common.markerIdentifiesObject(params.marker, vm))
+                    vms.push(vm);
```

Comment:
Please put braces around the statement for this multi-line if
statement (535-537).